python-lib/results/ab_statistics.py

```diff
@@ -0,0 +1,52 @@
+import pandas as pd
+
+from constants import Group, Columns
+
+
+class AbStatistics(object):
+    def __init__(self, user_reference_column: str, group_column: str, conversion_column: str):
+        self.user_reference_column = user_reference_column
+        self.group_column = group_column
+        self.conversion_column = conversion_column
+
+    def compute(self, results_df):
+        results_df = self.check_results_df(results_df)
+        aggregation = self.aggregate_by_group(results_df)
+        statistics_df = self.format_statistics(aggregation)
+        return statistics_df
+
+    def check_results_df(self, result_df):
```

Comment:
`check_reslults_df` should do what its name suggest: check and raise error message.
You can have another function called `filter_results_df` that returns the dataframe.
You should add logging here too as this is an important information: how many rows do we actually use.

Reply:
done in https://github.com/dataiku/dss-plugin-ab-testing/pull/2/commits/c9888533223d12828b49e2a8ca56a4030fb93a6d and https://github.com/dataiku/dss-plugin-ab-testing/pull/2/commits/76ac77dc1035450f736cdfe5a0978bd91ceeed85